### PR TITLE
refactor(anti-affinity-group): Migrate anti--affinity-group to egosca…

### PR DESCRIPTION
**Pull request by mlec1**
Migrate anti--affinity-group to egoscale v3

## Checklist
(For exoscale contributors)

* [ ] Changelog updated (under *Unreleased* block)
* [ ] Testing

## Testing

```
root@05ab6b31e301:/cli# go run . compute anti-affinity-group create test-mlec-2 --description "description"
 ✔ Creating Anti-Affinity Group "test-mlec-2"... 3s
┼─────────────┼──────────────────────────────────────┼
│ ID          │ f0b4b647-0be6-4d3d-b8ee-4c6202bbc071 │
│ Name        │ test-mlec-2                          │
│ Description │ description                          │
│ Instances   │ n/a                                  │
┼─────────────┼──────────────────────────────────────┼

root@05ab6b31e301:/cli# go run . compute anti-affinity-group list             
┼──────────────────────────────────────┼───────────┼
│                  ID                  │   NAME    │
┼──────────────────────────────────────┼───────────┼
│ aae67300-c277-44da-85a9-1b11023ea76a │ test-mlec │
┼──────────────────────────────────────┼───────────┼

root@05ab6b31e301:/cli# go run . compute anti-affinity-group show test-mlec

┼─────────────┼──────────────────────────────────────┼
│ ID          │ aae67300-c277-44da-85a9-1b11023ea76a │
│ Name        │ test-mlec                            │
│ Description │                                      │
│ Instances   │                                      │
┼─────────────┼──────────────────────────────────────┼

root@05ab6b31e301:/cli# go run . compute anti-affinity-group delete test-mlec
[+] Are you sure you want to delete Anti-Affinity Group test-mlec? [yN]: y
 ✔ Deleting Anti-Affinity Group test-mlec... 3s
root@05ab6b31e301:/cli# go run . compute anti-affinity-group delete f0b4b647-0be6-4d3d-b8ee-4c6202bbc071
[+] Are you sure you want to delete Anti-Affinity Group f0b4b647-0be6-4d3d-b8ee-4c6202bbc071? [yN]: y
 ✔ Deleting Anti-Affinity Group f0b4b647-0be6-4d3d-b8ee-4c6202bbc071... 3s

```

### CODE REVIEW

File: cmd/anti_affinity_group_create.go [hunk 1]
```diff
@@ -42,26 +39,43 @@ func (c *antiAffinityGroupCreateCmd) cmdPreRun(cmd *cobra.Command, args []string
...
+	if err != nil {
+		return err
+	}
+
+	antiAffinityGroup, err := antiAffinityGroupsResp.FindAntiAffinityGroup(c.Name)
+	if err != nil {
+		return err
+	}
+
 	return (&antiAffinityGroupShowCmd{
```
Comment (pierre-emmanuelJ) [inline cmd/anti_affinity_group_create.go:76]: same than other PR for `antiAffinityGroupShowCmd` type could be adapted here :)

File: cmd/anti_affinity_group_list.go [hunk 2]
```diff
@@ -45,22 +43,19 @@ func (c *antiAffinityGroupListCmd) cmdPreRun(cmd *cobra.Command, args []string)
...
+	antiAffinityGroups, err := globalstate.EgoscaleV3Client.ListAntiAffinityGroups(ctx)
 	if err != nil {
 		return err
 	}
 
 	out := make(antiAffinityGroupListOutput, 0)
 
-	for _, t := range antiAffinityGroups {
+	for _, t := range antiAffinityGroups.AntiAffinityGroups {
 		out = append(out, antiAffinityGroupListItemOutput{
```
Comment (pierre-emmanuelJ) [inline cmd/anti_affinity_group_list.go:56]: same  here for ID type

File: cmd/anti_affinity_group_list.go [hunk 3]
```diff
@@ -45,22 +43,19 @@ func (c *antiAffinityGroupListCmd) cmdPreRun(cmd *cobra.Command, args []string)
...
-			ID:   *t.ID,
-			Name: *t.Name,
+			ID:   t.ID.String(),
+			Name: t.Name,
```
Comment (pierre-emmanuelJ) [inline cmd/anti_affinity_group_list.go:58]: same  here for ID type, if not possible use `.String()` from ID

**Review by pierre-emmanuelJ — APPROVED** [review]

**Comment by pierre-emmanuelJ** [comment]
Great work 🎉

File: cmd/anti_affinity_group_list.go (hunk 3 above)
Comment (mlec1) [reply cmd/anti_affinity_group_list.go]: Here it was possible to remove the `.String()`

File: cmd/anti_affinity_group_create.go (hunk 1 above)
Comment (mlec1) [reply cmd/anti_affinity_group_create.go:76]: Here unfortunately, the `AntiAffinityGroup` needs to be a `string` in `antiAffinityGroupShowCmd`

**Review by pierre-emmanuelJ — APPROVED** [review]

**Review by pierre-emmanuelJ — APPROVED** [review]

**Review by elkezza — APPROVED** [review]
lgtm, good work : )

**Pull request merged**